File: app/src/main/java/uz/coder/davomatapp/todo/SharedPref.java
package uz.coder.davomatapp.todo;

import android.content.Context;
import android.content.SharedPreferences;

import uz.coder.davomatapp.R;

public class SharedPref {
    private static SharedPreferences sharedPreferences;

    private SharedPref(Context context) {
        sharedPreferences = context.getSharedPreferences(context.getString(R.string.app_name), android.content.Context.MODE_PRIVATE);
    }
    public static SharedPref getInstance(Context context) {
        return new SharedPref(context);
    }

    public void setBoolean(String key, boolean value) {
        sharedPreferences.edit().putBoolean(key, value).apply();
    }
    public boolean getBoolean(String key) {
        return sharedPreferences.getBoolean(key, false);
    }
    public void setString(String key, String value) {
        sharedPreferences.edit().putString(key, value).apply();
    }
    public String getString(String key) {
        return sharedPreferences.getString(key, "");
    }
    public void setInt(String key, int value) {
        sharedPreferences.edit().putInt(key, value).apply();
    }
    public int getInt(String key) {
        return sharedPreferences.getInt(key, 0);
    }
    public void setLong(String key, long value) {
        sharedPreferences.edit().putLong(key, value).apply();
    }
    public long getLong(String key) {
        return sharedPreferences.getLong(key, 0);
    }
    public void setFloat(String key, float value) {
        sharedPreferences.edit().putFloat(key, value).apply();
    }
    public float getFloat(String key) {
        return sharedPreferences.getFloat(key, 0);
    }
    public void clear(){
        sharedPreferences.edit().clear().apply();
    }
}

File: app/src/main/kotlin/uz/coder/davomatapp/todo/SharedPref.kt
package uz.coder.davomatapp.todo

import android.content.Context
import android.content.SharedPreferences
import uz.coder.davomatapp.R
import androidx.core.content.edit

class SharedPref private constructor(context: Context) {
    init {
        sharedPreferences =
            context.getSharedPreferences(context.getString(R.string.app_name), Context.MODE_PRIVATE)
    }

    fun setBoolean(key: String?, value: Boolean) {
        sharedPreferences.edit { putBoolean(key, value) }
    }

    fun getBoolean(key: String?): Boolean {
        return sharedPreferences.getBoolean(key, false)
    }

    fun setString(key: String?, value: String?) {
        sharedPreferences.edit { putString(key, value) }
    }

    fun getString(key: String?): String {
        return sharedPreferences.getString(key, "")!!
    }

    fun setInt(key: String?, value: Int) {
        sharedPreferences.edit { putInt(key, value) }
    }

    fun getInt(key: String?): Int {
        return sharedPreferences.getInt(key, 0)
    }

    fun setLong(key: String?, value: Long) {
        sharedPreferences.edit { putLong(key, value) }
    }

    fun getLong(key: String?): Long {
        return sharedPreferences.getLong(key, 0)
    }

    fun setFloat(key: String?, value: Float) {
        sharedPreferences.edit {putFloat(key, value)}
    }

    fun getFloat(key: String?): Float {
        return sharedPreferences.getFloat(key, 0f)
    }

    fun clear() {
        sharedPreferences.edit { clear() }
    }

    companion object {
        private lateinit var sharedPreferences: SharedPreferences

        fun getInstance(context: Context): SharedPref {
            return SharedPref(context)
        }
    }
}
